simulator/core/simulator.c

```diff
@@ -294,9 +295,11 @@ static void _shell(void) {
 				return _shell();
 			}
 		}
+		break;
```

Comment:
I don't think this `break` should be here, given the comment on line 279. Not the most intuitive control flow, but I think what that does is try to get a valid filename in the first half of the `r` case. If there is a file, then the second half of the `r` case will `return` (so fall through doesn't matter). If there isn't a file, then it relies on falling through the cases to get to the `h(elp)` case.

Reply:
OK, that makes sense, and yes, I see the fallthrough in the code (I almost wonder if a `goto` might help in making this cleaner? That's a problem for another time).